src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -1448,6 +1449,31 @@ void goto_instrument_parse_optionst::instrument_goto_program()
       throw 0;
   }
 
+  if(
+    cmdline.isset("replace-function-body") ||
+    cmdline.isset("generate-function-body"))
+  {
+    if(
```

Comment:
I have a mild preference to remove this code from here and put into the constructor of `replace_function_bodiest` so that the logic of how these parameters are handled is separated from the specific binary being used.

Reply:
I was actually the opposite; I had considered doing the parsing of the arguments outside of replace_function_bodiest entirely to avoid passing more "meaningful" strings around than we really have to

Reply:
My reason was that if someone wishes to re-use this they must copy/paste the interface to provide a consistent experience for the user. Rather than passing raw strings, what if you pass the whole `cmdlinet` into it? So it is clear they aren't meaningful strings but really the user input. 

The down side to my suggestion is becomes a class that is both responsible for generating functions and parsing command line args. Perhaps it would be better to have a third place that takes a cmdline and returns a `replace_goto_functionst`, but OK with leaving that to the first person who wishes to re-use this code.

Reply:
I'd like to support @thk123's position: In order to achieve what #1949 says, we will need this in all front-ends. So we really don't want to reimplement the command-line parsing multiple times. In my opinion, the last suggestion to have "a third place that takes a cmdline and returns..." is the best approach.

Reply:
Pushing the `only_generate` and `function_regex` logic into a reusable helper in `replace_function_bodies` would be preferable.